src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ParameterValidationAnalysis/ParameterValidationAnalysis.ParameterValidationDataFlowOperationVisitor.cs

```diff
@@ -352,8 +353,22 @@ private void ProcessRegularInvocationOrCreation(IMethodSymbol targetMethod, Immu
                         }
                     }
                 }
+
+                // Basic workaround to allow use of Guard fluent syntax as per https://github.com/dotnet/roslyn-analyzers/issues/3340
+                if (targetMethod.OriginalDefinition.Name == "NotNull" &&
```

Comment:
Potential additional checks:

1. Enclosing class is `static`
2. `NotNull` method is `static`
3. `Argument` method is `static`
4. `NotNull` parameter is of type `ArgumentInfo`.

Reply:
I could also add one more test for the first argument being of type `T` because we don't handle overload of the `Argument` method that takes an `Expression<Func<T>>` argument.